src/components/Receiving/ItemDetails.js

```diff
@@ -83,15 +83,18 @@ class ItemDetails extends Component {
       const piece = lineItems[item.poLineId].filter(el => el.id === item.id)[0];
 
       piece.isSelected = !piece.isSelected;
-      allChecked[item.poLineId] = false;
+
+      const isAllChecked = lineItems[item.poLineId].find(line => !line.isSelected || line.isSelected === false);
```

Comment:
it looks like `!line.isSelected` covers the second condition also, does it make sense to include both?

Reply:
Also naming looks weird, you have a var `isAllChecked`, but it will have `false` if all records are selected.